src/models/chart-model/format-pattern-from-range.js

```diff
@@ -0,0 +1,51 @@
+import KEYS from '../../constants/keys';
+
+export default function getFormatPatternFromRange(scaleName, viewHandler, layoutService) {
+  let min;
+  let max;
+  if (scaleName === KEYS.SCALE.X) {
+    ({ xAxisMin: min, xAxisMax: max } = viewHandler.getMeta().homeStateDataView);
+  } else {
+    ({ yAxisMin: min, yAxisMax: max } = viewHandler.getMeta().homeStateDataView);
+  }
+  if (!min || !max) {
+    return false;
+  }
+  let p = '';
+  const minMagnitude = Number(Number(min).toExponential().split('e')[1]);
```

Comment:
alternative without string manipulation
```suggestion
  const minMagnitude = Math.floor(Math.log10(Math.abs(min)));
```